pyupgrade.py

```diff
@@ -615,8 +664,17 @@ def _fix_tokens(contents_text, py3_plus):
 LENGTH_RE = re.compile('[hlL]?')
 
 
+def _must_match(regex, string, pos):
+    # type: (Pattern[Text], Text, int) -> Match[Text]
+    match = regex.match(string, pos)
+    assert match is not None
+    return match
+
+
 def parse_percent_format(s):
+    # type: (Text) -> Tuple[Tuple[Text, Optional[Fmt]], ...]
     def _parse_inner():
+        # type: () -> Iterator[Tuple[Text, Optional[Fmt]]]
```

Comment:
this one should be `Generator` as well

it is true that Generators are Iterators, but there are situations where an iterator cannot be used in place of a generator (for instance `@contextlib.contextmanager` or using `.send()`, etc.)